src/main/java/uk/gov/hmcts/reform/idam/web/AppController.java

```diff
@@ -396,26 +397,31 @@ public ModelAndView login(@ModelAttribute("authorizeCommand") @Validated Authori
-                    model.addAttribute(IS_ACCOUNT_SUSPENDED, true);
-                    bindingResult.reject("Account suspended");
-                } else if (errorCode == ErrorResponse.CodeEnum.POLICIES_FAIL) {
-                    log.info("/login: User failed policy checks - {}", obfuscateEmailAddress(request.getUsername()));
-                    model.addAttribute(HAS_POLICY_CHECK_FAILED, true);
-                    bindingResult.reject("Policy check failure");
-                    return new ModelAndView(LOGIN_VIEW, model.asMap());
-                } else if (errorCode == ErrorResponse.CodeEnum.STALE_USER_REGISTRATION_SENT) {
-                    Map<String, Object> staleUserResetPasswordParams = model.asMap();
-                    staleUserResetPasswordParams.remove(USERNAME);
-                    staleUserResetPasswordParams.remove(PASSWORD);
-                    staleUserResetPasswordParams.remove(SELF_REGISTRATION_ENABLED);
-                    return new ModelAndView("redirect:/reset/inactive-user", staleUserResetPasswordParams);
-                } else {
-                    model.addAttribute(HAS_LOGIN_FAILED, true);
-                    bindingResult.reject("Login failure");
+                switch(errorCode) {
+                    case ACCOUNT_LOCKED:
+                        model.addAttribute(IS_ACCOUNT_LOCKED, true);
+                        bindingResult.reject("Account locked");
+                    case ACCOUNT_SUSPENDED:
+                        model.addAttribute(IS_ACCOUNT_SUSPENDED, true);
+                        bindingResult.reject("Account suspended");
+                    case POLICIES_FAIL:
+                        log.info("/login: User failed policy checks - {}", obfuscateEmailAddress(request.getUsername()));
+                        model.addAttribute(HAS_POLICY_CHECK_FAILED, true);
+                        bindingResult.reject("Policy check failure");
+                        return new ModelAndView(LOGIN_VIEW, model.asMap());
+                    case STALE_USER_REGISTRATION_SENT:
+                        Map<String, Object> staleUserResetPasswordParams = model.asMap();
+                        staleUserResetPasswordParams.remove(USERNAME);
+                        staleUserResetPasswordParams.remove(PASSWORD);
+                        staleUserResetPasswordParams.remove(SELF_REGISTRATION_ENABLED);
+                        return new ModelAndView("redirect:/reset/inactive-user", staleUserResetPasswordParams);
+                    case ACCOUNT_LINKED_TO_EXTERNAL_PROVIDER:
+                        model.addAttribute(IS_ACCOUNT_SSO_ACCOUNT, true);
+                        bindingResult.reject("Account Linked to SSO Provider");
+                        return new ModelAndView(LOGIN_VIEW, model.asMap());
+                    default:
+                        model.addAttribute(HAS_LOGIN_FAILED, true);
+                        bindingResult.reject("Login failure");
```

Comment:
nice

Reply:
I believe this was left as a cascade of `if`s for a reason. The reason is you can't pass `null` to a `switch` statement and we want the `else` branch to cover both `null` and all non-matching values.

Reply:
why would it be null though, I feel like thats just bad from our end, having scenarios we dont cover

Reply:
I guess it's down to one's personal coding style. From my perspective, nothing ensures that it would **not** be `null`. There is no annotation saying this and if someone modifies the `ApiAuthResult` class then you suddenly have an uncovered scenario of `null` not being handled.